snippets/terraform/deployments/with-diagnostic-setting/main.tf

```diff
@@ -0,0 +1,57 @@
+terraform {
+  required_version = "~> 1.3"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.85"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+module "prerequisites" {
+  source   = "../../prerequisites"
+  location = var.location
+  name     = var.name
+  tags     = var.tags
+}
+
+data "azurerm_storage_account" "example" {
```

Comment:
at a later stage, we should just create these resources to give a realistic e2e example.

- create storage account.
- create deployment.
- create diagnostics.